lib/maze/request_list.rb

```diff
@@ -64,5 +64,19 @@ def clear
       @current = 0
       @count = 0
     end
+
+    # Sorts the first `count` elements of the list by the Bugsnag-Sent-At header, if present in all of those elements
+    def sort_by_sent_at!(count)
```

Comment:
Is there the potential for other threads to mess with the list while this is happening?

My knowledge is a bit limited, but I think only the server runs on a separate thread and that only appends. so it should be OK ?

Reply:
I believe this is the case as aside from the server and CLI services like docker and the interactive commands maze-runner should be based around a single thread.

Reply:
Yeah, it needed thinking about, but I think we're ok.  We need to account for the potential of more requests being added to the list, but only the MazeRunner (Cucumber) thread will move the current pointer on.  That's why the design includes the passing of the number of elements just received - it could break if it just sorted everything in the list every time.